services/streammanager/eventbridge/eventbridgemanager.go

```diff
@@ -37,14 +43,18 @@ func NewProducer(destinationConfig interface{}) (eventbridge.EventBridge, error)
 	if err != nil {
 		return eventbridge.EventBridge{}, fmt.Errorf("[EventBridge] Error while unmarshalling destination config :: %w", err)
 	}
+	httpClient := http.DefaultClient
+	httpClient.Timeout = o.Timeout
```

Comment:
Given that `http.DefaultClient` is essentially:
```go
var DefaultClient = &Client{}
```

It is not very good to set Timeout like this, cause it will override the global HTTP Default client (it is a pointer).

Thus is better to do:
```suggestion
	httpClient := http.Client{
	     Timeout: o.Timeout,
	}
```

Reply:
good catch
thanks!